lib/compute/agent-node-config.ts

```diff
@@ -6,108 +6,188 @@
  * compatible open source license.
  */
 
-import { CfnInstanceProfile, ServicePrincipal, Role } from '@aws-cdk/aws-iam';
-import { Fn, Stack, Tags } from '@aws-cdk/core';
-import { KeyPair } from 'cdk-ec2-key-pair';
-import { readFileSync } from 'fs';
-import { load } from 'js-yaml';
-import { JenkinsMainNode } from './jenkins-main-node';
 
-export interface AgentNodeProps {
-  amiId: string;
-  instanceType: string;
-  workerLabelString: string;
-  remoteUser: string;
-  initScript: string
-}
+ import * as iam from '@aws-cdk/aws-iam';
+ import { Fn, Stack, Tags, CfnParameter, Construct  } from '@aws-cdk/core';
+ import { KeyPair } from 'cdk-ec2-key-pair';
+ import { readFileSync } from 'fs';
+ import { load } from 'js-yaml';
+ import { JenkinsMainNode } from './jenkins-main-node';
+ 
+ 
+ export interface AgentNodeProps {
+   amiId: string;
+   instanceType: string;
+   workerLabelString: string;
+   remoteUser: string;
+   initScript: string
+ }
+ 
+ export interface AgentNodeNetworkProps {
+   readonly agentNodeSecurityGroup: string;
+   readonly subnetId: string;
+ }
+ 
+ export class AgentNodeConfig {
+   public readonly AgentNodeInstanceProfileArn: string;
+ 
+   public readonly STACKREGION: string;
+ 
+   private readonly ACCOUNT: string;
 
-export interface AgentNodeNetworkProps {
-  readonly agentNodeSecurityGroup: string;
-  readonly subnetId: string;
-}
+   public readonly SSHEC2KeySecretId: string;
+ 
+   constructor(stack: Stack) {
 
-export class AgentNodeConfig {
-  public readonly AgentNodeInstanceProfileArn: string;
+     this.STACKREGION = stack.region;
+     this.ACCOUNT = stack.account;
 
-  public readonly STACKREGION: string;
+     const agentAssumeRole = new CfnParameter(stack, 'agentAssumeRole', {
+      description: 'The assume role arn of the build agent',
+      allowedPattern: '.+',
+      default: false,
+     });
+     const assumeRole = stack.node.tryGetContext('agentAssumeRole');
 
-  public readonly SSHEC2KeySecretId: string;
+     const key = new KeyPair(stack, 'AgentNode-KeyPair', {
+       name: 'AgentNodeKeyPair',
+       description: 'KeyPair used by Jenkins Main Node to SSH into Agent Nodes',
+     });
+     Tags.of(key)
+       .add('jenkins:credentials:type', 'sshUserPrivateKey');
+     const AgentNodeRole = new iam.Role(stack, 'OpenSearch-CI-AgentNodeRole', {
```

Comment:
import Role construct and just use that. 
e.g., `import { Role } from '@aws-cdk/aws-iam'`
          `new Role(x, y, z);`